src/toil/lib/threading.py

```diff
@@ -533,6 +533,9 @@ def leave(self) -> Iterator[bool]:
                 # There is someone claiming to be here. Are they alive?
                 full_path = os.path.join(self.lockfileDir, item)
 
+                if not os.path.exists(full_path): # suddenly file doesnt exist on network file system?
+                    continue
+
```

Comment:
This isn't going to 100% close the race here between opening the file and someone deleting it. We probably want to try/catch around the open instead.

Reply:
Thanks for getting back to me so fast and excellent point! How does it look now?

Reply:
Hey @adamnovak I hope you are well. Was hoping could get this snippet reviewed and then merged? We love toil and are using it as our workflow engine!

Reply:
Oh sorry, I didn't notice this was updated!